config/package_resouce.go

```diff
@@ -14,26 +14,229 @@
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_INSTALLED:
+			p.policy.MSI.Install = pr
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_REMOVED:
+			p.policy.MSI.Remove = pr
+		}
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_Googet:
+		pr := p.GetGooget()
+		if !packages.GooGetExists {
+			return nil, fmt.Errorf("cannot manage GooGet package %q because googet does not exist on the system", pr.GetName())
+		}
+
+		switch p.GetDesiredState() {
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_INSTALLED:
+			p.policy.GooGet.Install = pr
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_REMOVED:
+			p.policy.GooGet.Remove = pr
+		}
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_Yum:
+		pr := p.GetYum()
+		if !packages.YumExists {
+			return nil, fmt.Errorf("cannot manage Yum package %q because yum does not exist on the system", pr.GetName())
+		}
+
+		switch p.GetDesiredState() {
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_INSTALLED:
+			p.policy.Yum.Install = pr
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_REMOVED:
+			p.policy.Yum.Remove = pr
+		}
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_Zypper_:
+		pr := p.GetZypper()
+		if !packages.ZypperExists {
+			return nil, fmt.Errorf("cannot manage Zypper package %q because zypper does not exist on the system", pr.GetName())
+		}
+
+		switch p.GetDesiredState() {
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_INSTALLED:
+			p.policy.Zypper.Install = pr
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_REMOVED:
+			p.policy.Zypper.Remove = pr
+		}
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_Rpm:
+		pr := p.GetRpm()
+		if !packages.RPMExists {
+			return nil, fmt.Errorf("cannot manage RPM package because rpm does not exist on the system")
+		}
+
+		switch p.GetDesiredState() {
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_INSTALLED:
+			p.policy.RPM.Install = pr
+		case agentendpointpb.ApplyConfigTask_Config_Resource_PackageResource_REMOVED:
+			p.policy.RPM.Remove = pr
+		}
+	default:
+		return nil, errors.New("SystemPackage field not set or references unknown package manager")
+	}
+
+	return &ManagedResources{Packages: []ManagedPackage{p.policy}}, nil
+}
+
+// TODO: implement a caching system for installed packages.
+var aptInstalled []packages.PkgInfo
+var debInstalled []packages.PkgInfo
+var gooInstalled []packages.PkgInfo
+var yumInstalled []packages.PkgInfo
+var zypperInstalled []packages.PkgInfo
+var rpmInstalled []packages.PkgInfo
+
+func needsInstall(installedPkgs []packages.PkgInfo, name string) bool {
+	for _, pkg := range installedPkgs {
+		if pkg.Name == name {
+			return false
+		}
+	}
+	return true
 }
 
 func (p *packageResouce) checkState() (bool, error) {
```

Comment:
Would it be better to rename the method or have a named return value so that it is clear what the boolean values mean?
https://g3doc.corp.google.com/go/g3doc/style/decisions.md?cl=head#named-result-parameters

Currently, it is not clear what true or false return values mean.

Reply:
Done, also renamed function and reworked logic to make it clear.

Reply:
Did you push the changes. Still seeing the old method name.

Reply:
That was a comment for the wrong function. Added in the named return